libmate-desktop/mate-desktop-item.c

```diff
@@ -2012,8 +2017,9 @@ ditem_execute (const MateDesktopItem *item,
 					     sn_context);
 		sn_launcher_context_unref (sn_context);
 	}
+	if (GDK_IS_X11_DISPLAY (display))
+		sn_display_unref (sn_display);
```

Comment:
in local rpm build.
```
In function 'ditem_execute',
    inlined from 'mate_desktop_item_launch_on_screen_with_env' at mate-desktop-item.c:2131:8:
mate-desktop-item.c:2021:17: warning: 'sn_display' may be used uninitialized [-Wmaybe-uninitialized]
 2021 |                 sn_display_unref (sn_display);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mate-desktop-item.c: In function 'mate_desktop_item_launch_on_screen_with_env':
mate-desktop-item.c:1779:20: note: 'sn_display' was declared here
 1779 |         SnDisplay *sn_display;
      |                    ^~~~~~~~~~
```

Reply:
See
https://github.com/mate-desktop/mate-desktop/pull/587
which initialises `sn_display` to NULL in the wayland case instead of ignoring it

Reply:
But this on is still there in local build and in travis-ci builds (debian and fedora), now at line 2024
https://app.travis-ci.com/github/mate-desktop/mate-desktop/jobs/611266565#L1826
https://app.travis-ci.com/github/mate-desktop/mate-desktop/jobs/611266566#L1069 
```
mate-desktop-item.c:2024:3: warning: 1st function call argument is an uninitialized value [core.CallAndMessage]

                sn_display_unref (sn_display);

                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
This is in a different function so unaffected by the first round of changes in this PR. Note that `sn_display` should only be uninitialized here if not running in x11, and in that case this code should not be reached. In fact, cppcheck didn't find it, but Travis does. Force-pushing with this fixed as well.

Reply:
I saw this `-Wmaybe-uninitialized` warnings with gcc compiler flags from `-Wall`.
That we use cppcheck for debian CI is really questionable for me, because nobody from the team use it and fix those tons of warnings which are only spams debian build log.
Anyway, warnings are gone.
Thanks for fixing them.